### ISSUE TITLE
Osm_way_id speeds do not match osm tag speed or default speed of highway type

### ISSUE THREAD

**Issue by subrina0013** [matsim-org/matsim-code-examples#1035]
I am trying to understand why the osm_way_ids in my network do not match with what is in https://github.com/matsim-org/matsim-libs/blob/97f655596ca6961d12f268cde1bec9bd0389ce85/contribs/osm/src/main/java/org/matsim/contrib/osm/networkReader/LinkProperties.java#L30-L33.  For example: I have residential links that have different speed limits where it should be 15 km/hr? Why is that? If the osm ids have different speed, then what is the point of setting the default speed? And based on what the speed is changing?

![image](https://github.com/matsim-org/matsim-code-examples/assets/42855310/c90e3e44-91ac-43e1-8441-252cb9d2ccda)

**Comment by Janekdererste** [comment]
The freespeed is set [here](https://github.com/matsim-org/matsim-libs/blob/97f655596ca6961d12f268cde1bec9bd0389ce85/contribs/osm/src/main/java/org/matsim/contrib/osm/networkReader/SupersonicOsmNetworkReader.java#L254) during the conversion process which calls [this](https://github.com/matsim-org/matsim-libs/blob/97f655596ca6961d12f268cde1bec9bd0389ce85/contribs/osm/src/main/java/org/matsim/contrib/osm/networkReader/SupersonicOsmNetworkReader.java#L294) method which tries to parse the osm speed tag. 

The freespeed is then set to something lower than the original speed tag [here](https://github.com/matsim-org/matsim-libs/blob/4c624687b6d8a93864522fbfc7b2b6bbc2303274/contribs/osm/src/main/java/org/matsim/contrib/osm/networkReader/LinkProperties.java#L162) in case it is an urban link. The heuristic to decide whether it is an urban link is European centric, as it simply tests for a speed lower than 50km/h. The freespeed is reduced to account for things like traffic lights and intersections in urban contexts, as these things are usually not explicitly modelled in MATSim. 

For 'non-urban' links the freespeed is taken from the osm tag. 

In case we don't have a speed tag in osm, the default speed from the link properties you have found is used [here](https://github.com/matsim-org/matsim-libs/blob/4c624687b6d8a93864522fbfc7b2b6bbc2303274/contribs/osm/src/main/java/org/matsim/contrib/osm/networkReader/LinkProperties.java#L182). This method again applies some heuristic to reduce the speed on links the converter considers 'urban links'.

**Comment by subrina0013** [comment]
So, step1: original speed is obtained from osm tag 
step2: check if it is urban or not. If urban then reduce, if not then original osm tag
step3: calculate for unknown speed tag with length of that link.
so, where in these steps the assigned speed limits (from MATSim) stands? only for calculating if the original speed is unknown?

![image](https://github.com/matsim-org/matsim-code-examples/assets/42855310/b280d5e9-9175-4162-b3dd-23a703dd5cf4)

**Comment by Janekdererste** [comment]
Yes, the freespeed from `LinkProperties` is only used in case there is no speed limit tag in osm.
